src/main/resources/mybatis/features.xml

```diff
@@ -15,6 +15,30 @@
                               st_setsrid(geom, 4326)
                              )
         </if>
+        <if test="countries != null">
+             and country_cd in <foreach item="i" collection="countries" open="(" separator="," close=")">#{i}</foreach>
+         </if>
+        <if test="states != null">
+            and country_cd || ':' || state_cd in <foreach item="i" collection="states" open="(" separator="," close=")">#{i}</foreach>
+        </if>
+        <if test="counties != null">
+            and country_cd || ':' || state_cd || ':' || county_cd in <foreach item="i" collection="counties" open="(" separator="," close=")">#{i}</foreach>
+        </if>
+        <if test="nationalAquiferCode != null">
+            and nat_aqfr_cd = #{nationalAquiferCode}
+        </if>
+        <if test="monitoringLocationNumber != null">
+            and monitoring_location.site_identification_number  = #{monitoringLocationNumber}
+        </if>
+        <if test="agencyCode != null">
+            and agency_cd = #{agencyCode}
+        </if>
+        <if test="hydrologicUnits != null">
+            and calculated_huc_12 in <foreach item="i" collection="hydrologicUnits" open="(" separator="," close=")">#{i}</foreach>
+        </if>
+        <if test="monitoringLocationType != null">
+            and site_type in <foreach item="i" collection="monitoringLocationType" open="(" separator="," close=")">#{i}</foreach>
+        </if>
```

Comment:
Is this safe from XSS and efficient for query speed?

Reply:
This is run entirely on the server end. cross-site scripting does not come into play.
The should be fairly efficient once there is an index on  monitoringLocationType (another ticket).